cou/utils/juju_utils.py

```diff
@@ -301,14 +325,15 @@ async def run_on_unit(
         :type timeout: Optional[int]
         :returns: action.data['results'] {'Code': '', 'Stderr': '', 'Stdout': ''}
         :rtype: Dict[str, str]
-        :raises: UnitNotFound
-        :raises: ActionFailed
+        :raises UnitNotFound: _description_
+        :raises ActionFailed: _description_
```

Comment:
provide a simple description

Reply:
You are right, I tried to be fast and did not realized that it's `ApplicationFailed` and not `ApplicationNotFound`.

Reply:
What do you say about this `When the application status is in error (it's not 'completed').`?

Reply:
Sure, let's try to be consistent and change to this description on all `ApplicationFailed`